service/raft/raft_group0.cc

```diff
@@ -461,6 +463,7 @@ future<> raft_group0::join_group0(std::vector<gms::inet_address> seeds, bool as_
                 // In a fresh cluster this will trigger an empty snapshot transfer which is redundant but correct.
                 // See #14066.
                 nontrivial_snapshot = true;
+                is_discovery_leader = true;
```

Comment:
Can `is_discovery_leader` be unified with `nontrivial_snapshot`?

Reply:
I agree with Kamil, and that was the reason I kept them separate.

Reply:
IMO if we will only leave  `is_discovery_leader` it will even clarify the code. It will be clear that only if a node is discovery_leader nontrivial snapshot is required.

Reply:
This is no longer relevant in v2, as the bool was removed.